### DISCUSSION TITLE
Revert force push

### DISCUSSION THREAD

**Discussion by chriskoenigsfeld** [community/community#42683] [Repositories]
### Select Topic Area

Product Feedback

### Body

I have a problem. I created a markdown file in my repository in github. That means in my browser. At the same time I coded in the same project and pushed my commits. And because it said I had differences to the origin and since I am the only one working on my repo I force pushed it. And after a while, when I wanted to read my markdown file again, it was empty. Just then I remembered, what I did and why there was a warning. I overwrite my changes of this markdown file with my local commits.

I hope it is possible to get my old history or commits from the remote history back, before I force pushed my local commits and overwrite the remote history with my local history.
I hope you understand what I mean and I hope it is possible to get the old history by github-tools or github-admins back, because I never had these remote changes on my local history, and that's why for example reflog doesn't work on this problem.

**Comment by jfullstackdev** [comment]
did you commit using PR ? because you can simply revert it on GitHub Web if that is through PR, without that, you will be dealing with Git

try GitHub Desktop too, clone it and look at the History, right click it and simply click revert

**Comment by chriskoenigsfeld** [comment]
Okay I tried GitHub Desktop. But I can only see the new history. How can I get the old one with the overwritten commits?

**Reply by jfullstackdev** [reply]
not very sure of that because what I'm thinking is that, you overwritten it in the local, so there is no way to actually get that history

if you push that on GitHub, it must be there

**Reply by chriskoenigsfeld** [reply]
No. I made changes in github itself and changes in my local repo. Then I force pushed my local changes and overwrite the changes on github

**Comment by chriskoenigsfeld** [comment]
Hope this makes it clearer, what my problem is :-/
![github-problem](https://user-images.githubusercontent.com/94907836/209704691-c51100e2-891b-4aec-a848-d7f63a52de7f.png)

**Comment by spenserblack** [comment]
It is possible to access and interact with unreachable commits (e.g. `git fsck --unreachable/--lost-found`). The issue is *finding* the commit. Unfortunately, you force-pushed over a commit that exists only in the remote repository. Maybe GitHub support would be able to find the SHA for you, then you can at least view the diff on the web and redo your work. For example, here's an unreachable commit in a remote repo (note that I believe GitHub may prune unreachable commits that aren't linked to a PR): https://github.com/spenserblack/test-repo/commit/fe8f948746628d466f00b2916c3f76c8a6bf8387
